src/niMidi.cpp

```diff
@@ -234,21 +311,58 @@ class NiMidiSurface: public BaseSurface {
 			this->_sendSysex(CMD_TRACK_NAME, 0, numInBank, name);
 			// todo: level meters, volume, pan
 		}
-		// todo: navigate tracks, navigate banks
+		// todo: navigate tracks, navigate banks. NOTE: probably not here
+	}
+
+	void ClearSelected() {
+		// Clear all selected tracks. Copyright (c) 2010 and later Tim Payne (SWS)
+		int iSel = 0;
+		for (int i = 0; i <= GetNumTracks(); i++) // really ALL tracks, hence no use of CSurf_NumTracks
+			GetSetMediaTrackInfo(CSurf_TrackFromID(i, false), "I_SELECTED", &iSel);
```

Comment:
Note that if we're going to use GetNumTracks, we should probably use GetTrack, not CSurf_TrackFromID.

Reply:
Hmmm, you may be right. At least judging from the function name. I just copied from SWS without questioning. But as stated above, my understanding is the GetNumTracks is not zero based: If you have a master track plus 2 tracks GetNumTracks yields 2. The loop unselects master and these 2 tracks.
Side Note: Contrary to the 4D encoder which currently cannot select the master track the top row button at the very left can!